app/controllers/products_controller.rb

```diff
@@ -0,0 +1,66 @@
+class ProductsController < ApplicationController
+
+before_action only: [:show, :edit, :update, :retire] { @product = Product.find(params[:id]) }
+
+  def index
+    if params[:search]
+      @products = Product.search(params[:search]).where(retire: false).order(stock: :desc)
+    else
+    @products = Product.where(retire: false).order(stock: :desc)
+    end
+  end
+
+  def show
+    @review = Review.new(product_id: @product.id)
+    if @order_item = @current_order.orderitems.where("product_id = ?", params[:id]).first
+    else
+      @order_item = Orderitem.new(product_id: @product.id)
+    end
+  end
+
+  def category
+    @category = Category.find_by(name: params[:category_name])
+    @products = @category.products.where(retire: false)
+  end
+
+  def robot
+    @robot = Robot.find(params[:id])
+    @products = @robot.products.where(retire: false)
+  end
+
+  def new
+    @product = Product.new
+  end
+
+  def create
+    @product = Product.new(product_params)
+    if @product.save
+      redirect_to robot_path(current_robot)
+    else
+      render 'new'
+    end
+  end
+
+  def edit; end
+
+  def update
+    @product.update(product_params)
```

Comment:
Calling the `update` method will do the `save` as part of the `update`, so i think it would make sense to have the `update` as the condition in the `if` rather than calling `save` again